src/Fractal/FieldsModifier.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Giadc\JsonApiResponse\Fractal;
+
+use Giadc\JsonApiRequest\Requests\RequestParams;
+use Giadc\JsonApiResponse\Interfaces\AbstractDataModifier;
+
+class FieldsModifier implements AbstractDataModifier
+{
+    public function __invoke(array $data, string $resourceKey, RequestParams $requestParams): array
```

Comment:
To prevent our client and API from getting out of sync, it would be great if a validation error was thrown if the client requests/excludes fields that do not exist.